storage/series/series.go

```diff
@@ -161,22 +152,29 @@ func (s *dbSeries) shouldExpire(now, blockStart time.Time) bool {
 }
 
 func (s *dbSeries) updateBlocksWithLock() {
-	now := s.opts.ClockOptions().NowFn()()
-	allBlocks := s.blocks.AllBlocks()
+	var (
+		now             = s.opts.ClockOptions().NowFn()()
+		allBlocks       = s.blocks.AllBlocks()
+		allBlocksSealed = true
+	)
 	for blockStart, block := range allBlocks {
 		if s.shouldExpire(now, blockStart) {
 			s.blocks.RemoveBlockAt(blockStart)
 			block.Close()
+		} else if block.IsSealed() {
+			continue
 		} else if s.shouldSeal(now, blockStart, block) {
 			block.Seal()
+		} else {
+			allBlocksSealed = false
 		}
 	}
+	if allBlocksSealed {
+		s.blocks.Seal()
+	}
```

Comment:
Can't do this as discussed due to the `shouldSeal` method needing to be taken into account.